kornia/enhance/adjust.py

```diff
@@ -943,3 +971,33 @@ def __init__(self, brightness_factor: Union[float, torch.Tensor]) -> None:
 
     def forward(self, input: torch.Tensor) -> torch.Tensor:
         return adjust_brightness(input, self.brightness_factor)
+
+
+class Invert(nn.Module):
+    r"""Inverts the values of an input tensor by its maximum value.
+
+    Args:
+        input (torch.Tensor): The input tensor to invert with an arbitatry shape.
+        max_val (torch.Tensor): The expected maximum value in the input tensor. The shape has to
+          according to the input tensor shape, or at least has to work with broadcasting. Default: 1.0.
+
+    Example:
+        >>> img = torch.rand(1, 2, 4, 4)
+        >>> Invert()(img).shape
+        torch.Size([1, 2, 4, 4])
+
+        >>> img = 255. * torch.rand(1, 2, 3, 4, 4)
+        >>> Invert(torch.tensor(255.))(img).shape
+        torch.Size([1, 2, 3, 4, 4])
+
+        >>> img = torch.rand(1, 3, 4, 4)
+        >>> Invert(torch.tensor([[[[1.]]]]))(img).shape
+        torch.Size([1, 3, 4, 4])
+    """
+
+    def __init__(self, max_val: torch.Tensor = torch.tensor(1.)) -> None:
+        super(Invert, self).__init__()
+        self.max_val = max_val
```

Comment:
Currently alright. I am thinking to have:

```python
if not isinstance(max_val, nn.Parameter): 
    self.register_buffer("max_val", max_val)
else:
    self.max_val = max_val
```

To make it more appropriate.

Reply:
+1